components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -337,7 +342,23 @@ void AtmosphereOutput::init()
 
     // Store the field layout, so that calls to setup_output_file are easier
     const auto& layout = fid.get_layout();
-    m_vars_dims[fname] = get_var_dimnames(layout);
+    m_vars_dims[fname] = get_var_dimnames(m_transpose ? layout.transpose() : layout);
+
+    // Initialize a helper_field for each unique layout.  This can be used for operations
+    // such as writing transposed output.
+    if (m_transpose) {
+      const auto helper_layout = layout.transpose();
+      const std::string helper_name   = "transposed_"+helper_layout.to_string();
+      if (m_helper_fields.find(helper_name) == m_helper_fields.end()) {
+        // We can add a new helper field for this layout
+        using namespace ekat::units;
+        FieldIdentifier fid_helper(helper_name,helper_layout,Units::invalid(),fid.get_grid_name());
```

Comment:
I wonder if we should create different helper fields for different scalar types. In case of average output, with avg tracking enabled, we probably want to also transpose the avg count field before writing it. But that field has data type `IntType`. So maybe we should append `"_" + e2str(fid.data_type())` to `helper_name`, and create `fid_helper` with the data type from `fid`.